File: app/src/main/java/com/example/github_api_concrete/model/data/remote/GitHubAPI.java
package com.example.github_api_concrete.model.data.remote;

import com.example.github_api_concrete.model.pojo.pullrequests.Response;
import com.example.github_api_concrete.model.pojo.repos.RepositoriesResult;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface GitHubAPI {

    @GET("search/repositories")
    Observable<RepositoriesResult> getAllRepositories(@Query("q") String language,
                                                      @Query("sort") String sort,
                                                      @Query("page") int page);

    @GET("repos/{owner}/{repo}/pulls")
    Observable<List<Response>> getAllPRs(@Path("owner") String owner,
                                         @Path("repo") String repo);
}

File: app/src/main/java/com/hako/githubapi/data/repository/retrofit/GithubApi.kt
package com.hako.githubapi.data.repository.retrofit

import com.hako.githubapi.domain.entities.PullRequest
import com.hako.githubapi.domain.entities.Repositories
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GithubApi {

    @GET("/search/repositories")
    fun getTopRepositories(
        @Query("q") language: String,
        @Query("sort") sort: String,
        @Query("page") page: Int
    ): Observable<Repositories>

    @GET("/repos/{author}/{name}/pulls")
    fun getPullRequests(
        @Path("author") author: String,
        @Path("name") name: String
    ): Observable<List<PullRequest>>
}
